### PR TITLE
[INTEG-1417] Update aiig create upload function to handle latest upload function for both spaceId and environmentId

### PR DESCRIPTION
**Pull request by ryunsong-contentful**
## Purpose
The create asset upload function has been changed to take both spaceId and the environmentId, therefore we need to adjust accordingly to the new api params.

## Approach

* Pass environment id when creating the upload
* Check the created upload and return the environment scoped URL when the environment is present. (Preserves backward compatibility when CMA.js doesn't have support for environment ids in uploads.)

## Testing steps

Testing results on sandbox:

Correct images: 
image

Using environment scoped upload URLs:

image

## Dependencies and/or References
Will require latest contentful-management.js package version or else merging this PR will break the backend

### CODE REVIEW

**Comment by jsdalton** [comment]
Currently blocked during testing by https://github.com/contentful/app-upload-to-delivery/pull/269

**Comment by jsdalton** [comment]
Currently blocked by https://github.com/contentful/node-apps-toolkit/pull/461

File: apps/ai-image-generator/backend/src/helpers/upload-images.ts [hunk 1]
```diff
@@ -10,11 +11,17 @@ const UPLOAD_DOMAIN: Record<string, URL> = {
   ),
 };
 
+type UploadPropsWithEnvironment = {
+  // TODO: use upstream UplpadProps directly once environment id has been added
```
Comment (whitelisab) [inline apps/ai-image-generator/backend/src/helpers/upload-images.ts:15]: nit: typo in this comment, `UplpadProps`should be `UploadProps` 😄

**Review by whitelisab — APPROVED** [review]
Looks good! 🎊

**Pull request merged**